deploy/docker/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi7-minimal
+FROM registry.access.redhat.com/ubi8-minimal
```

Comment:
This needs a large discussion before we merge this. Moving from UBI 7 and UBI 8 should not be taken lightly. We need to find out what ramifications this has.

Reply:
Yes I agree, but as far as I am concerned, ubi7 does not support arm64...I am wondering if the cross build would work, if so the change of base image can be removed.

Reply:
I see this:

```bash
$ docker buildx imagetools inspect registry.access.redhat.com/ubi7-minimal:latest
Name:      registry.access.redhat.com/ubi7-minimal:latest                                                                                                                                        
MediaType: application/vnd.docker.distribution.manifest.list.v2+json                                                                                                                             
Digest:    sha256:8c74fc2c392e6816049068ef1bcd71068eaf40ae6d94b233c5c98d9129e17c84              
                                                
Manifests:                                                                                                                                                                                       
  Name:      registry.access.redhat.com/ubi7-minimal:latest@sha256:adf8765f5a564df0888c441d4c285eb34800ab175e079c93581870741e6954c3                                                              
  MediaType: application/vnd.docker.distribution.manifest.v2+json                               
  Platform:  linux/amd64    
                                                                                                
  Name:      registry.access.redhat.com/ubi7-minimal:latest@sha256:4902f66bb401bd4b2171b7ab0c267c24ce21e7bd4debc749d55210baa25a8197
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/ppc64le
              
  Name:      registry.access.redhat.com/ubi7-minimal:latest@sha256:6e1e81408e49b1e0f54c6bb23ce9b385ea543ecd1258b844825b20860cd220a8
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/s390x
```

ARM64 is not listed for UBI7. So, looks like @MrXinWang is right and this change is probably required to move forward with builds for ARM64.

I don't think this move is that bad. In #1115 we moved from centos7 to UBI7. Then in #1199 we moved from UBI7 to UBI7-minimal. In both cases, only sanity checks were done. So, I think a sanity check is all we need in order to merge this.

Reply:
moving to UBI8 may (IIRC) cause this to not be able to run on Red Hat 7.x systems. I will test that out to see.

Reply:
@jmazzitelli Thanks!

Reply:
Smoke test on my end did not show any problems. But I remember hearing people (I think it was during OLM/Operator discussions) that UBI8 images cannot run or are not supported on RHEL 7.x platforms. So we need to research this more. If UBI8 is not supported on RHEL 7 platforms, then we need to discuss this PR further because we can't lose support on RHEL 7 systems.

Reply:
Any suggestions for me to do for researching this problem? :)))) Is there a convenient way to setup an development environment of RHEL 7 to test this?

Reply:
Maybe we can have two images one for rhel7 and another for rhel8

Reply:
I will ask around about the UBI8 / RHEL issue. I don't have any details, just something I thought I remember hearing. I'll report back.

Reply:
I was correct in my original thinking.

https://access.redhat.com/support/policy/rhel-container-compatibility

I confirmed that the RHEL team has said they will support UBI7 images on top of RHEL8 host, but the reverse is not necessarily true. While it is believed that most UBI8 images would work on a RHEL7 host, those images are not officially supported by RHEL. So if there is a problem the responsibility falls on the container owner to work it out.

So, IMO, this is a no-go if the amd64 images produced by our builds are based on UBI8.

Perhaps think about two Dockerfiles and somehow switching between them (when ARM, use the Dockerfile-UBI8, when amd, use Dockerfile-UBI7). That's just one thought. Not sure if there are other possible solutions. But we can't currently have the amd64 image based on UBI8.

Reply:
Goof to know

IMO in any moment we'll move to rhel8 so maybe is not a bad idea have 2 dockerfile and start to test rhel8. What do you think?

Reply:
I think having two Dockerfiles - one for UBI7 and one for UBI8 - would be nice. We should have a flag to switch between the two (rather than the GOARCH - define an env var "KIALI_DOCKER_FILE" - the default should use UBI7 if amd64 and the default should use UBI8 if ARM, but you should be able to override by passing the env var in which would allow us to build amd64 Kiali using UBI8 so we can test it). This would do all of that - put this in the root `Makefile` after the line that defines GO_BUILD_ENVVARS:

```
ifeq ($(GOARCH),amd64)
KIALI_DOCKER_FILE ?= Dockerfile-ubi7-minimal
else
KIALI_DOCKER_FILE ?= Dockerfile-ubi8-minimal
endif
```

If KIALI_DOCKER_FILE env var is already defined, it will be used (that's what the "?=" does).

Then, where ever we run the `docker build` commands, we need to pass in `-f ${KIALI_DOCKER_FILE}`.

Reply:
great @jmazzitelli :)

Reply:
@jmazzitelli Great solution indeed! I have updated the code, please review again. Thanks!